### PR TITLE
BUGFIX: ONE-3476 Colors prop override colorPalette from style settings

### PR DESCRIPTION
**Pull request by ZdenekVidensky**

### CODE REVIEW

**Comment by ZdenekVidensky** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2394/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1801/ : FAILURE in 3m 38s

**Comment by ZdenekVidensky** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2396/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1802/ : SUCCESS in 2m 57s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/954/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2397/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/556/ : UNSTABLE in 23m 04s

**Comment by ZdenekVidensky** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2407/ : SUCCESS in 12s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1810/ : SUCCESS in 3m 31s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2408/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1811/ : SUCCESS in 2m 53s

**Comment by ZdenekVidensky** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/959/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2409/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/560/ : SUCCESS in 14m 13s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2413/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1814/ : FAILURE in 51s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2414/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1815/ : SUCCESS in 1m 55s

File: src/components/uri/tests/Visualization.spec.tsx [hunk 1]
```diff
@@ -445,35 +446,127 @@ describe('VisualizationWrapped', () => {
...
+                const BaseChartElement = wrapper.find(BaseChart).get(0);
+                expect(BaseChartElement.props.config.colorPalette).toEqual(expectedColorPalette);
+                expect(mutatedSdk.project.getColorPaletteWithGuids).toHaveBeenCalledTimes(0);
+            });
+        });
 
-        return testUtils.delay(SLOW + 1).then(() => {
-            wrapper.update();
-            expect(mutatedSdk.project.getColorPaletteWithGuids).toHaveBeenCalledTimes(1);
+        it('should call getColorPaletteWithGuids from sdk and render BaseChart with this palette', () => {
```
Comment (ivan-nejezchleb) [inline src/components/uri/tests/Visualization.spec.tsx]: ... when no colors/palette in props

File: src/components/uri/tests/Visualization.spec.tsx [hunk 2]
```diff
@@ -445,35 +446,127 @@ describe('VisualizationWrapped', () => {
...
+                fetchVisualizationClass,
+                uriResolver,
+                intl
+            };
+
+            const wrapper = mount(
+                <VisualizationWrapped {...props as any}/>
+            );
+
+            return testUtils.delay(SLOW + 1).then(() => {
```
Comment (ivan-nejezchleb) [inline src/components/uri/tests/Visualization.spec.tsx]: use FAST delay everywhere

File: src/components/uri/tests/Visualization.spec.tsx [hunk 3]
```diff
@@ -445,35 +446,127 @@ describe('VisualizationWrapped', () => {
...
+                wrapper.update();
+                expect(wrapper.find(BaseChart).length).toBe(1);
```
Comment (ivan-nejezchleb) [inline src/components/uri/tests/Visualization.spec.tsx]: redundant expect, not needed here. Remove in all new cases

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2415/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1816/ : SUCCESS in 2m 47s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2417/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1818/ : SUCCESS in 1m 48s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2419/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1820/ : SUCCESS in 3m 09s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/512/ : SUCCESS in 12s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1827/ : SUCCESS in 6m 51s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/963/ : UNSTABLE in 33m 29s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/467/ : SUCCESS in 24m 03s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/967/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2430/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/566/ : SUCCESS in 20m 01s

**Comment by ZdenekVidensky** [comment]
ok to test

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/515/ : SUCCESS in 18s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1833/ : SUCCESS in 2m 23s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/968/ : UNSTABLE in 26m 03s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/470/ : SUCCESS in 16m 33s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/518/ : SUCCESS in 23s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1836/ : SUCCESS in 4m 24s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/971/ : UNSTABLE in 27m 35s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/473/ : SUCCESS in 21m 44s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/519/ : SUCCESS in 16s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1843/ : SUCCESS in 2m 46s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/972/ : SUCCESS in 21m 50s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/474/ : SUCCESS in 21m 08s

**Pull request merged**